packages/x-data-grid-pro/src/hooks/features/serverSideData/serverSideInterfaces.ts

```diff
@@ -0,0 +1,63 @@
+import { GridRowId } from '@mui/x-data-grid';
+import { GridGetRowsParams, GridGetRowsResponse } from '../../../models';
+
+/**
+ * The dataSource API interface that is available in the grid [[apiRef]].
+ */
+export interface GridDataSourceApi {
```

Comment:
What are the use-cases for the APIs here?

Reply:
Apart from the internal use (like being used in NestedDataManager), mostly customizations on the userland.

A few examples:
1. `fetchTopLevelRows` could be used to implement a `Refetch root data` button outside the grid.
2. `enqueueChildrenFetch` and `gridServerSideDataErrorsSelector` could be used to implement `Refetch failed nodes` outside the Grid.

Makes sense?

P.S. Some of these customizations could even be added in the docs.